File: app/src/main/java/com/example/soccernews/data/services/FavoriteDao.java
package com.example.soccernews.data.services;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.soccernews.data.model.News;

import java.util.List;

@Dao
public interface FavoriteDao {

    @Query("SELECT * FROM News")
    LiveData<List<News>> getFavoriteNews();

    @Query("SELECT id FROM News")
    List<Integer> getFavoriteId();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(News news);

    @Delete
    void delete(News news);
}

File: app/src/main/java/viana/alison/soccernews/data/service/FavoriteDao.kt
package viana.alison.soccernews.data.service

import androidx.room.*
import kotlinx.coroutines.flow.Flow
import viana.alison.soccernews.data.model.News

@Dao
interface FavoriteDao {

    @Query("SELECT * FROM News")
    fun getFavoriteNews(): Flow<List<News>>

    @Query("SELECT id FROM News")
    fun getFavoriteIds(): Flow<List<Int>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(news: News)

    @Delete
    suspend fun delete(news: News)

}